### PR TITLE
docs(self-managed): updates for release cycle v8.4

### PR DESCRIPTION
**Pull request by HamzaMasood1**
## Description

Task: https://github.com/camunda/distribution/issues/164
## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [x] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [ ] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by christinaausley** [comment]
@Zelldon Are you the best person to have a peek here?

**Comment by Zelldon** [comment]
Thanks @christinaausley I can take a look.

Hey @HamzaMasood1 thanks for creating this PR, I think one important part we miss is that this is a breaking change for Zeebe. Afaik I understood the contribution here https://github.com/camunda/zeebe/pull/15462, is that users who have a different installation than with K8 and helm need to do manual steps, like changing the ownership (right?). See discussion https://github.com/camunda/zeebe/pull/15462#discussion_r1416745067, especially the note https://github.com/camunda/zeebe/pull/15462#discussion_r1420396025

P.S.: I'm not a fan of this change.

\cc @npepinpe @megglos

**Review by npepinpe — CHANGES_REQUESTED** [review]
Hi @HamzaMasood1, thanks for the PR! What you wrote explains the change well, but doesn't tell users what possible impact this has on existing deployments, and what they can do to resolve those/have a smooth update experience.

As @Zelldon mentioned, we need to highlight the fact this is a breaking change and a necessary step to update from 8.3 to 8.4 for anyone not using the official Helm charts. 

We should highlight:

- [ ] The impact of the change for those running bare metal or Docker deployments (e.g. file permission issues)
- [ ] A step by step/runbook on how to update as smoothly as possible.

**Comment by HamzaMasood1** [comment]
Thank for the feedback guys. I'll try my best.

**Comment by aabouzaid** [comment]
I've added a `hold` label so the PR is not merged till we add all changes related to the upgrade.

File: docs/self-managed/platform-deployment/helm-kubernetes/upgrade.md [hunk 1]
```diff
@@ -93,6 +93,15 @@ For more details on the Keycloak upgrade path, you can also read the [Bitnami Ke
 
 ## Version update instructions
 
+### v8.4.0
```
Comment (christinaausley) [inline docs/self-managed/platform-deployment/helm-kubernetes/upgrade.md:96]: Is there any other related documentation we should be noting this in given the breaking changes?

Comment (aabouzaid) [reply docs/self-managed/platform-deployment/helm-kubernetes/upgrade.md:96]: I will work on this just after updating the parts related to the Helm chart versioning change.

Comment (aabouzaid) [reply docs/self-managed/platform-deployment/helm-kubernetes/upgrade.md]: Done :heavy_check_mark:

**Comment by aabouzaid** [comment]
@christinaausley This PR is ready for review :heavy_check_mark:

**Review by christinaausley — APPROVED** [review]

**Comment by aabouzaid** [comment]
@npepinpe I believe that Hamza addressed the 2 points you have mentioned.
So if there is nothing else left, I will merge the PR after your final review.

**Pull request merged**